kite-tools-parent/kite-tools/src/main/java/org/kitesdk/cli/commands/DeleteCommand.java

```diff
@@ -46,11 +51,24 @@ public int run() throws IOException {
         View view = Datasets.load(uriOrName);
         Preconditions.checkArgument(viewMatches(view.getUri(), uriOrName),
             "Resolved view does not match requested view: " + view.getUri());
-        view.deleteAll();
+
+        if(console.isDebugEnabled()){
+          for(PartitionView<?> pv: (Iterable<PartitionView<?>>) view.getCoveringPartitions()){
+            console.debug("Deleting partition {}", pv.getLocation());
```

Comment:
I guess I'm not following, are you saying getCoveringPartitions() doesn't return a partition boundary?  Is this because if someone gave a URI with a constraint you didn't partition on it would still be part of the URI?